### PR TITLE
Fixup a bunch of miscellaneous issues.

### PR DESCRIPTION
**Pull request by doublej472**
I looked back through these docs again after my last PR and found a few issues
and awkward grammar, so this new PR should fix them up.

Most of these changes are adding a period to the end of sentences, and changing some awkward or misleading phrases, and adding detail for why you might want a specific option.

I'm not actually sure how fast a player moves with respect to `baby-zombie-movement-speed`, I made up a value by judging how fast a baby zombie moves relative to me walking in a paper server. It's not obvious from the decompiled code how speed is calculated.

### CODE REVIEW

File: source/server/configuration.rst [hunk 1]
```diff
@@ -81,23 +81,23 @@ use-versioned-world
...
+* **description**: Instructs the server to return a list of players when
   tab-completing if the plugin has no tab completions of its own.
 
 enable-player-collisions
 ~~~~~~~~~~~~~~~~~~~~~~~~
 * **default**: true
 * **description**: Sets whether the server should allow players to collide with
   one another.
-* **warning**: This setting can be broken by plugins interacting with the
-  scoreboard, double check plugins when troubleshooting this value.
```
Comment (zachbr) [inline source/server/configuration.rst]: Not sure why this was moved down to the wrong config option?

File: source/server/configuration.rst [hunk 2]
```diff
@@ -81,23 +81,23 @@ use-versioned-world
...
 
 save-empty-scoreboard-teams
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 * **default**: false
 * **description**: Some scoreboard plugins leave hundreds of empty scoreboard
   teams around, dramatically slowing down login times. This sets whether the
   server should remove those empty teams automatically.
+* **warning**: This setting can be broken by plugins interacting with the
+  scoreboard, double check plugins when troubleshooting this value.
```
Comment (zachbr) [inline source/server/configuration.rst]: See above comment on `enable-player-collisions`

File: source/server/configuration.rst [hunk 3]
```diff
@@ -523,7 +525,7 @@ spawner-nerfed-mobs-should-jump
 baby-zombie-movement-speed
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 * **default**: 0.5
-* **note**: For reference, the vanilla player moves with a speed of 0.1.
+* **note**: For reference, the vanilla player moves with a speed of ~0.4.
```
Comment (zachbr) [inline source/server/configuration.rst]: This whole option is wrong as of newer versions. It's a modifier, express as a percentage.
So `0.5` is 50% faster than the base mob speed. and `-0.4` would be 40% slower than the base mob speed.

I'll be renaming this option to `baby-zombie-movement-modifier` shortly.

File: source/server/configuration.rst (hunk 1 above)
Comment (doublej472) [reply source/server/configuration.rst]: I forgot the reason why I did it before, but it's definitely wrong. I've moved it back to the correct location, thanks for catching that!

Relevant issue: https://github.com/PaperMC/Paper/issues/2517

File: source/server/configuration.rst (hunk 3 above)
Comment (doublej472) [reply source/server/configuration.rst]: I've just updated this PR to reflect the new config name, and used your wording to describe what it does.

**Comment by doublej472** [comment]
@zachbr I've fixed the issues you mentioned, let me know what you think!

File: source/server/configuration.rst [hunk 4]
```diff
@@ -520,11 +522,13 @@ spawner-nerfed-mobs-should-jump
...
-baby-zombie-movement-speed
-~~~~~~~~~~~~~~~~~~~~~~~~~~
+baby-zombie-movement-modifier
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 * **default**: 0.5
-* **note**: For reference, the vanilla player moves with a speed of 0.1.
-* **description**: Controls the speed baby zombies move at.
+* **description**: Modifies the speed that baby zombies move at, where 0.5 is
+  50% faster than the mob base speed, and -0.4 would be 40% slower.
+* **note**: For reference, a non-sprinting player moves with a speed modifer
```
Comment (zachbr) [inline source/server/configuration.rst]: Player does not have a speed modifier. In looking for obvious references, I'm not sure we even need one. I would just remove this note line. It seems unnecessary now that it's a modifier and the 'base' speed/reference speed is that of the zombie itself.

Comment (doublej472) [reply source/server/configuration.rst]: That makes sense, I suppose people will know how fast a baby zombie moves normally as a reference. I just pushed a change that addresses this!

**Pull request merged**